### PR TITLE
Upgrade to new dependency mockito-testng

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Upgrade to new dependency mockito-testng
Fixied issue in code related to this changes in mockito
- thenThrow(Class) should only throw checked exceptions that match the signature of the stubbed method (https://github.com/mockito/mockito/issues/1155)
- Cleaner code for Mockito users by better integration with static code checkers, more @CheckReturnValue (https://github.com/mockito/mockito/issues/1270)
- Add @CheckReturnValue to stubbing/verification methods (https://github.com/mockito/mockito/issues/1228)
- Removed usused stubs from test classes (stubbed methods that would never be invoked during test executions).
- In some cases, stubs would be counted as unused, when one is declared for multiple tests in @BeforeMethod, and at least one of the tests ends up not invoking the stubbed method. Usual solution to that is disabling strict checking for that stub using static method lenient(), or moving/removing stubs.
- In case when same method would be stubbed with when().then() (but with different arguments), an exception will occur about unnecessary duplicated. To fix that, stubbing were reworked into doReturn().when().
- In other cases, when stubbing is intentionally declared, but not invoked, a lenient() method is used.

Merge state

### What issues does this PR fix or reference?
https://github.com/mockito/mockito/issues/962
https://github.com/eclipse/che-parent/pull/77
https://github.com/eclipse/che/issues/10617

Upgrade to official  mockito-testng artifact 
#### Release Notes

Upgrade to official  mockito-testng artifact 

#### Docs PR
n/a

### CODE REVIEW

File: wsagent/che-core-git-impl-jgit/src/test/java/org/eclipse/che/git/impl/jgit/JGitConnectionTest.java [hunk 1]
```diff
@@ -192,7 +191,8 @@ public void shouldDoNothingWhenTransportHttpReceived() throws Exception {
      * with  help real children {@link TransportHttp}, which returns real not null
      * value. And then we can create mock {@link TransportHttp}.
      */
-    mock(Transport.class);
```
Comment (garagatyi) [inline wsagent/che-core-git-impl-jgit/src/test/java/org/eclipse/che/git/impl/jgit/JGitConnectionTest.java:195]: Is this change needed in this PR?

File: infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/multiuser/oauth/IdentityProviderConfigBuilderTest.java [hunk 2]
```diff
@@ -225,7 +225,8 @@ public void testRethrowOnInvalidTokenBadRequestException() throws Exception {
 
   @Test(expectedExceptions = {InfrastructureException.class})
   public void testRethrowOnAnyException() throws Exception {
-    when(keycloakServiceClient.getIdentityProviderToken(anyString())).thenThrow(Exception.class);
```
Comment (garagatyi) [inline infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/multiuser/oauth/IdentityProviderConfigBuilderTest.java:228]: Can you elaborate on why we need this change? I don't see any connection with the purpose of the PR.

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntimeTest.java [hunk 3]
```diff
@@ -357,7 +357,7 @@ public void stopsWaitingAllMachineStartWhenOneMachineStartFailed() throws Except
     final ImmutableMap<String, Pod> allPods =
         ImmutableMap.of(WORKSPACE_POD_NAME, mockPod(ImmutableList.of(container1, container2)));
     when(k8sEnv.getPods()).thenReturn(allPods);
-    doThrow(InfrastructureException.class).when(bootstrapper).bootstrapAsync();
+    doThrow(IllegalStateException.class).when(bootstrapper).bootstrapAsync();
```
Comment (garagatyi) [inline infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntimeTest.java:398]: Can you elaborate on why we need this change?

File: infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntimeTest.java [hunk 4]
```diff
@@ -378,7 +378,7 @@ public void throwsInfrastructureExceptionWhenErrorOccursAndCleanupFailed() throw
     doNothing().doThrow(InfrastructureException.class).when(namespace).cleanUp();
     when(k8sEnv.getServices()).thenReturn(singletonMap("testService", mock(Service.class)));
     when(services.create(any())).thenThrow(new InfrastructureException("service creation failed"));
-    doThrow(InfrastructureException.class).when(namespace).services();
```
Comment (garagatyi) [inline infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntimeTest.java:419]: Can you elaborate on why we need this change? I don't see any connection with the purpose of the PR.

**Review by garagatyi — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Comment by musienko-maxim** [comment]
ci-test

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/935/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/935//Selenium_20tests_20report/)
docker image: eclipseche/che-server:10647
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/936/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/936//Selenium_20tests_20report/)
docker image: eclipseche/che-server:10647
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/937/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/937//Selenium_20tests_20report/)
docker image: eclipseche/che-server:10647
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by mkuznyetsov** [comment]
MockitoTestNG runs with enabled Strict Stubbing rules, so tests were adjusted to that:

1. Removed usused stubs from test classes (stubbed methods that would never be invoked during test executions).
2. In some cases, stubs would be counted as unused, when one is declared for multiple tests in @BeforeMethod, and at least one of the tests ends up not invoking the stubbed method. Usual solution to that is disabling strict checking for that stub using static method `lenient()`.
3. In case when same method would be stubbed with `when().then()` (but with different arguments), an exception will occur about unnecessary duplicated. To fix that, stubbing were reworked into `doReturn().when()`.
4. In other cases, when stubbing is intentionally declared, but not invoked, a `lenient()` method is used.

More info on how Strict Stubbing checking workes at:
https://static.javadoc.io/org.mockito/mockito-core/2.7.1/org/mockito/exceptions/misusing/PotentialStubbingProblem.html

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/953/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/953//Selenium_20tests_20report/)
docker image: eclipseche/che-server:10647
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.
